python/hawkey/subject-py.c

```diff
@@ -287,6 +288,45 @@ reldep_possibilities_real(_SubjectObject *self, PyObject *args, PyObject *kwds)
     return possibilitiesToPyObject(iter, sack);
 }
 
+static PyObject *
+get_best_query(_SubjectObject *self, PyObject *args, PyObject *kwds)
+{
+    PyObject *sack;
+    DnfSack *csack;
+    PyObject *forms = NULL;
+    PyObject *with_nevra = NULL;
+    PyObject *with_provides = NULL;
+    PyObject *with_filenames = NULL;
+    const char *kwlist[] = {"sack", "with_nevra", "with_provides", "with_filenames", "forms", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O!|O!O!O!O", (char**) kwlist, &sack_Type, &sack,
+        &PyBool_Type, &with_nevra, &PyBool_Type, &with_provides,
+        &PyBool_Type, &with_filenames, &forms)) {
+        return NULL;
+    }
+    HyForm *cforms = NULL;
+    if ((forms != NULL) && (forms != Py_None) &&
+        ((!PyList_Check(forms)) || (PyList_Size(forms) > 0))) {
+        cforms = fill_form(forms);
+        if (cforms == NULL)
+            return NULL;
+    }
+    gboolean c_with_nevra = with_nevra == NULL || PyObject_IsTrue(with_nevra);
+    gboolean c_with_provides = with_provides == NULL || PyObject_IsTrue(with_provides);
+    gboolean c_with_filenames = with_filenames == NULL || PyObject_IsTrue(with_filenames);
+    csack = sackFromPyObject(sack);
+    HyNevra nevra = NULL;
+    HyQuery query = hy_subject_get_best_solution(self->pattern, csack, cforms, &nevra, self->icase,
+                                                 c_with_nevra, c_with_provides, c_with_filenames);
+
+    PyObject *py_query = queryToPyObject(query, sack, &query_Type);
+    if (nevra != NULL) {
+        g_clear_pointer(&nevra, hy_nevra_free);
```

Comment:
Replace this line (g_clear_pointer) by: `hy_nevra_free(nevra);`
"nevra" is local variable. We do not need set its value to NULL because it is not used anymore.